torch/ao/pruning/semi_structured_sparse_tensor.py

```diff
@@ -0,0 +1,131 @@
+import torch
+
+
+class SemiStructuredSparseTensor(torch.Tensor):
+    """
+    This class implementes 2x4 sparsity
+    """
+
+    @staticmethod
+    def __new__(
+        cls,
+        original_tensor,
+        original_shape=None,
+        compressed_tensor=None,
+        cslt=None,
+        transposed=False,
+        contiguous_output=False,
```

Comment:
I don't think contiguous_output should be a user setting. The contiguity is a contract tied to PyTorch operators. If the output of an operation is usually contiguous, then when passing the structured sparse layout it should also be contiguous.

Reply:
Contiguous mode is quite a bit slower than the non-contiguous mode, so I think it's worth having some way to specify which one you want. 

![image](https://github.com/pytorch/pytorch/assets/8041643/bcb57209-e01a-4d64-bdd8-7502ab93011e)

This can live somewhere else off of the tensor though.